instrumentation/jsonrpc4j-1.3/testing/src/main/java/io/opentelemetry/instrumentation/jsonrpc4j/v1_3/AbstractJsonRpcTest.java

```diff
@@ -0,0 +1,101 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.instrumentation.jsonrpc4j.v1_3;
+
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.assertThat;
+import static io.opentelemetry.sdk.testing.assertj.OpenTelemetryAssertions.equalTo;
+import static io.opentelemetry.semconv.incubating.RpcIncubatingAttributes.RPC_JSONRPC_ERROR_CODE;
+import static io.opentelemetry.semconv.incubating.RpcIncubatingAttributes.RPC_JSONRPC_VERSION;
+import static io.opentelemetry.semconv.incubating.RpcIncubatingAttributes.RPC_METHOD;
+import static io.opentelemetry.semconv.incubating.RpcIncubatingAttributes.RPC_SERVICE;
+import static io.opentelemetry.semconv.incubating.RpcIncubatingAttributes.RPC_SYSTEM;
+import static java.nio.charset.StandardCharsets.UTF_8;
+
+import com.googlecode.jsonrpc4j.JsonRpcBasicServer;
+import io.opentelemetry.api.trace.SpanKind;
+import io.opentelemetry.instrumentation.testing.junit.InstrumentationExtension;
+import io.opentelemetry.testing.internal.jackson.databind.JsonNode;
+import io.opentelemetry.testing.internal.jackson.databind.ObjectMapper;
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.TestInstance;
+
+@SuppressWarnings("deprecation") // using deprecated semconv
+@TestInstance(TestInstance.Lifecycle.PER_CLASS)
+public abstract class AbstractJsonRpcTest {
+
+  protected abstract InstrumentationExtension testing();
+
+  protected abstract JsonRpcBasicServer configureServer(JsonRpcBasicServer server);
+
+  @Test
+  void testServer() throws IOException {
+    CalculatorService calculator = new CalculatorServiceImpl();
+    JsonRpcBasicServer server =
+        configureServer(new JsonRpcBasicServer(calculator, CalculatorService.class));
+
+    JsonNode response =
+        testing()
+            .runWithSpan(
+                "parent",
+                () -> {
+                  InputStream inputStream =
+                      new ByteArrayInputStream(
+                          "{\"jsonrpc\":\"2.0\",\"method\":\"add\",\"params\":[1,2],\"id\":1}"
+                              .getBytes(UTF_8));
+                  ByteArrayOutputStream outputStream = new ByteArrayOutputStream();
+                  server.handleRequest(inputStream, outputStream);
```

Comment:
this test does not verify that context is propagated, perhaps should use a real http request?